grafana/deploy.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# This is a super simple approach to syncing dashboards
+# TODO:
+#  - check if dashboard really needs update (will cause many versions in grafana db)
+#  - check if stale folder needs to be deleted
+#  - check if stale dashboard needs to be deleted 
+
+set -eu
+
+export RESOURCEGROUP=${GLOBAL_RESOURCEGROUP:-global}
+
+if [[ -z ${GRAFANA_NAME} ]];
+then
+    echo "GRAFANA_NAME needs to be set"
+    exit 1
+fi
+
+cd dashboards
+
+existing_folders=$(az grafana folder list -g ${RESOURCEGROUP}  -n ${GRAFANA_NAME})
+
+IFS=$'\n'; for d in $(ls -1)
+do
+    if [[ $(echo $existing_folders | jq --arg TITLE "$d"  -c '.[] | select( .title == $TITLE )' |wc -l )  -gt 0 ]];
+    then
+        folderUid=$(echo $existing_folders | jq --arg TITLE "$d"  -r '.[] | select( .title == $TITLE ) | .uid' )
+    else
+        if [[ ${DRY_RUN} == "true" ]];
+        then
+            echo "would create folder '${d}' on managed grafana ${GRAFANA_NAME} in rg ${RESOURCEGROUP}"
+        else
+            folderUid=$(az grafana folder create --only-show-errors  -g ${RESOURCEGROUP}  -n ${GRAFANA_NAME} --title ${d} |jq -r '.uid')
+        fi
+    fi
+    cd ${d}
+    IFS=$'\n'; for dashboard in $(ls -1)
+    do
+        dashboard_name=$(cat ${dashboard} | jq '.dashboard.title' )
+        if [[ ${DRY_RUN} == "true" ]];
+        then
+            echo "would create/update dashboard '${dashboard_name}' on managed grafana ${GRAFANA_NAME} in rg ${RESOURCEGROUP}"
+        else
+            sed -i "s/XYZTOBESETBYPIPELINEZYX/${folderUid}/" ${dashboard}
```

Comment:
sed will not error if this magic string is not found (and replaced). Maybe we should first check for this magic string and error out if it is not found? In case someone forgets to set it in the dashboard json...

```suggestion
            grep -q XYZTOBESETBYPIPELINEZYX ${dashboard} || echo "Magic string XYZTOBESETBYPIPELINEZYX not found in dashboard file ${dashboard} >&2
            sed -i "s/XYZTOBESETBYPIPELINEZYX/${folderUid}/" ${dashboard}
```